rules/jsonapi-document-structure-resource-objects.js

```diff
@@ -0,0 +1,316 @@
+// Document Structure - Resource Objects - https://jsonapi.org/format/#document-resource-objects
+
+// All rules in the file MUST have corresponding tests
+
+import { enumeration, length, truthy } from '@stoplight/spectral-functions';
+
+export default {
+  documentationUrl: 'https://jsonapi.org/format/#document-resource-objects',
+  rules: {
+
+    /**
+     * Ensures that a `Resource Object` is of type `object` or `array`
+     */
+    'document-structure-resource-objects-type': {
+      description: 'A `Resource Object` MUST be of type `object` or `array`',
+      message: `{{path}} - {{description}}`,
+      severity: 'error',
+      // given: "$..[?(@.get || @.delete || @.put || @.patch || @.post)]..[?(@.responses || @.requestBody)]..content['application/vnd.api+json'].schema.properties.data.properties",
```

Comment:
Should this comment be removed?

Reply:
Removed comment. Old JSONPath Expression not in use, provided a more explicit JSONPath Expression.